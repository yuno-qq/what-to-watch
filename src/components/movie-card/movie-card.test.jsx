import React from 'react'
import {describe, test, expect} from '@jest/globals'
import renderer from 'react-test-renderer'

import MovieCard from './movie-card.jsx'


describe(`<MovieCard> snapshots:`, () => {
  test(`should render movie "Красный дракон"`, () => {
    const movie = {
      imageSrc: `https://placehold.co/280x175/EEE/31343C`,
      url: `/`,
      previewSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
      name: `Красный дракон`
    }

    const tree = renderer.create(<MovieCard movie={movie} onMouseEnter={() => {}} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
