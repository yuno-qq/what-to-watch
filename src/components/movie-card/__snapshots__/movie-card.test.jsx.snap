// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard> snapshots: should render movie "Красный дракон" 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      loop={true}
      muted={true}
      onCanPlay={[Function]}
      onWaiting={[Function]}
      poster="https://placehold.co/280x175/EEE/31343C"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/"
    >
      Красный дракон
    </a>
  </h3>
</article>
`;
